xdsl/irdl/constraints.py

```diff
@@ -652,6 +660,20 @@ def mapping_type_vars(
             tuple(c.mapping_type_vars(type_var_mapping) for c in self.param_constrs),
         )
 
+    def __or__(self, value: GenericAttrConstraint[_AttributeCovT], /):
+        if (
+            not isinstance(value, ParamAttrConstraint)
+            or self.base_attr is not cast(ParamAttrConstraint[Any], value).base_attr
```

Comment:
```suggestion
            or self.base_attr is not cast(ParamAttrConstraint[Attribute], value).base_attr
```
would this fix the need for the `ignore`?

Reply:
Nope for whatever reason Pyright is not happy with Attribute:

```
Type "Attribute" cannot be assigned to type variable "ParametrizedAttributeCovT@ParamAttrConstraint"
  Type "Attribute" is not assignable to upper bound "ParametrizedAttribute" for type variable "ParametrizedAttributeCovT@ParamAttrConstraint"
    "Attribute" is not assignable to "ParametrizedAttribute"Pyright[reportInvalidTypeArguments](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportInvalidTypeArguments)
```